features/docs/writing_support_code/hooks/README.md

```diff
@@ -0,0 +1,123 @@
+# Cucumber Hooks
+
+Cucumber proposes several hooks to let you specify some code to be executed at
+different stage of the test execution, like before or after the execution of a
+scenario.
+
+Hooks are part of your support code.
+
+They are executed in the following order:
+
+- [AfterConfiguration](#afterconfiguration-and-installplugin)
+- [InstallPlugin](#afterconfiguration-and-installplugin)
+- [BeforeAll](#beforeall-and-afterall)
+  - Per scenario:
+    - [Around](#around)
+    - [Before](#before-and-after)
+      - Per step:
+        - [AfterStep](#afterstep)
+    - [After](#before-and-after)
+- [AfterAll](#beforeall-and-afterall)
+
+You can define as many hooks as you want. If you have several hooks of the same
+types - for example, several `BeforeAll` hooks - they will be all executed once.
```

Comment:
`<new para>`

Multiple hooks of the same type are executed in the order that they were defined. If you wish to control this order, use manual requires in `env.rb` (This file is loaded first), or migrate them all to one `hooks.rb` file.